management/src/main/java/nearsoft/skytouch/management/service/Impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,28 @@
+package nearsoft.skytouch.management.service.Impl;
+
+import nearsoft.skytouch.common.model.Product;
+import nearsoft.skytouch.management.repository.ProductRepository;
+import nearsoft.skytouch.management.service.ProductService;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+class ProductServiceImpl implements ProductService {
+
+    private ProductRepository productRepository;
+
+    public ProductServiceImpl(ProductRepository productRepository) {
```

Comment:
How are you giving a value to productRepository?

Reply:
ProductServiceImpl is annotated as a Repository so it is injected here

Reply:
Annotating a class as a repository turns that class into a bean in the application context, so that springboot can manage it, to make us of springboot's dependency injection you should be autowiring that repository on this class.

Reply:
I think it is no necessary use the Autowire annotation in this case, as you can see here https://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-spring-beans-and-dependency-injection.html
if there are only  one constructor that annotation it's not needed

Reply:
I would still use the annotation, just for clarity, however, that is up to you then. Thank you.